File: backend/app/schemas/matching.py
from pydantic import BaseModel
from typing import Optional, List, Dict
from datetime import datetime

class MatchPreferenceBase(BaseModel):
    skill_weight: Optional[float] = 0.3
    availability_weight: Optional[float] = 0.2
    style_weight: Optional[float] = 0.2
    goals_weight: Optional[float] = 0.3
    preferred_meeting_frequency: Optional[int] = None
    preferred_session_duration: Optional[int] = None
    timezone: Optional[str] = None
    language_preferences: Optional[List[str]] = None
    industry_preferences: Optional[List[str]] = None
    custom_criteria: Optional[Dict] = None
    exclusions: Optional[Dict] = None

class MatchPreferenceCreate(MatchPreferenceBase):
    pass

class MatchPreferenceUpdate(MatchPreferenceBase):
    pass

class MatchScoreBase(BaseModel):
    mentor_id: int
    mentee_id: int
    total_score: float
    skill_match_score: float
    availability_score: float
    style_match_score: float
    goals_alignment_score: float
    match_details: Dict
    status: str

class MatchScoreCreate(MatchScoreBase):
    pass

class MatchScoreUpdate(BaseModel):
    status: Optional[str] = None
    match_details: Optional[Dict] = None

class MatchScore(MatchScoreBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

class MatchFeedbackBase(BaseModel):
    rating: int
    feedback: Optional[str] = None
    suggestions: Optional[str] = None

class MatchFeedbackCreate(MatchFeedbackBase):
    match_id: int
    user_id: int

class MatchFeedback(MatchFeedbackBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: frontend/src/types/matching.ts
export interface MatchPreference {
    skill_weight: number;
    availability_weight: number;
    style_weight: number;
    goals_weight: number;
    preferred_meeting_frequency?: number;
    preferred_session_duration?: number;
    timezone?: string;
    language_preferences?: string[];
    industry_preferences?: string[];
    custom_criteria?: Record<string, any>;
    exclusions?: Record<string, any>;
}

export interface MatchScore {
    id: number;
    mentor_id: number;
    mentee_id: number;
    total_score: number;
    skill_match_score: number;
    availability_score: number;
    style_match_score: number;
    goals_alignment_score: number;
    match_details: {
        matching_skills: string[];
        matching_availability: number;
        [key: string]: any;
    };
    status: "suggested" | "accepted" | "rejected" | "active";
    created_at: string;
    updated_at: string;
}

export interface Match {
    mentor: {
        id: number;
        name: string;
        rating?: number;
        total_sessions?: number;
    };
    mentee: {
        id: number;
        name: string;
    };
    compatibility_score: number;
    common_interests: string[];
    status: 'pending' | 'accepted' | 'rejected';
    match_score: MatchScore;
}

export interface MatchFeedback {
    rating: number;
    feedback?: string;
    suggestions?: string;
}

export interface MentorProfile {
    user_id: number;
    name: string;
    bio?: string;
    experience_years?: number;
    company?: string;
    position?: string;
    expertise_areas: string[];
    mentoring_style?: string;
    availability: AvailabilitySlot[];
}

export interface MenteeProfile {
    user_id: number;
    name: string;
    bio?: string;
    goals?: string;
    current_position?: string;
    desired_skills: string[];
    learning_style?: string;
    availability: AvailabilitySlot[];
}

export interface AvailabilitySlot {
    id: number;
    start_time: string;
    end_time: string;
    recurrence?: string;
    is_available: boolean;
}
